bgpd/bgp_zebra.c

```diff
@@ -3628,6 +3628,8 @@ static int bgp_zebra_srv6_sid_notify(ZAPI_CALLBACK_ARGS)
 			bgp_vrf->vpn_policy[AFI_IP6].tovpn_sid_locator = locator;
 			bgp_vrf->vpn_policy[AFI_IP6].tovpn_sid_transpose_label =
 				label;
+			UNSET_FLAG(bgp_vrf->vpn_policy[AFI_IP6].flags,
+				   BGP_VPN_POLICY_TOVPN_SID_REQUEST_IN_PROGRESS);
```

Comment:
Shouldn't we unset the `BGP_VPN_POLICY_TOVPN_SID_REQUEST_IN_PROGRESS ` flag also when SID allocation fails (case `ZAPI_SRV6_SID_FAIL_ALLOC`)?

Reply:
actually, SID allocation failure is not the culprit; all is related to the decision: which nexthop to use, and when should we consider to invalidate exported vpn prefixes.